internal/driver/glfw/window.go

```diff
@@ -582,12 +592,25 @@ func (w *window) mouseMoved(viewport *glfw.Window, xpos float64, ypos float64) {
 
 	if w.cursor != cursor {
 		// cursor has changed, store new cursor and apply change via glfw
+		newCursor, isStdCursor := fyneToNativeCursor(cursor)
 		w.cursor = cursor
-		if cursor == nil {
+		oldCursor := w.rawCursor
+
+		if isStdCursor {
+			w.rawCursor = nil
+		} else {
+			w.rawCursor = newCursor
+		}
+
+		if newCursor == nil {
 			viewport.SetInputMode(glfw.CursorMode, glfw.CursorHidden)
 		} else {
 			viewport.SetInputMode(glfw.CursorMode, glfw.CursorNormal)
-			viewport.SetCursor(cursor)
+			viewport.SetCursor(newCursor)
+		}
+		if oldCursor != nil {
+			// destroy old cursor now as to avoid viewport reverting to default cursor even for an instant
+			oldCursor.Destroy()
```

Comment:
I guess, this could be simplified to this:
```go
		rawCursor, isStdCursor := fyneToNativeCursor(cursor)
		if rawCursor == nil {
			viewport.SetInputMode(glfw.CursorMode, glfw.CursorHidden)
		} else {
			viewport.SetInputMode(glfw.CursorMode, glfw.CursorNormal)
			viewport.SetCursor(cursor)
			viewport.SetCursor(rawCursor)
		}

		if c.rawCursor != nil {
			// destroy old cursor now as to avoid viewport reverting to default cursor even for an instant
			c.rawCursor.Destroy()
			w.rawCursor = nil
		}
		if !isStdCursor {
			w.rawCursor = rawCursor
		}
```